homeassistant/components/sensor/phicomm.py

```diff
@@ -150,79 +155,67 @@ def __init__(self, hass, username, password):
         old_devs = self.devs
-        self.update_data()
+        await self.update_data()
 
         tasks = []
-        for device in self._devices:
+        for device in self.devices:
             if device.state != device.state_from_devs(old_devs):
                 _LOGGER.info('%s: => %s', device.name, device.state)
                 tasks.append(device.async_update_ha_state())
 
         if tasks:
-            yield from asyncio.wait(tasks, loop=self._hass.loop)
+            await asyncio.wait(tasks, loop=self._hass.loop)
 
-    def update_data(self):
+    async def update_data(self):
         """Update online data."""
         try:
-            json = self.fetch_data()
+            json = await self.fetch_data()
             if ('error' in json) and (json['error'] != '0'):
                 _LOGGER.debug("Reset token: error=%s", json['error'])
                 self._token = None
-                json = self.fetch_data()
+                json = await self.fetch_data()
             self.devs = json['data']['devs']
             _LOGGER.info("Get data: devs=%s", self.devs)
         except BaseException:
+            self.devs = {}
             import traceback
             _LOGGER.error('Exception: %s', traceback.format_exc())
 
-    def fetch_data(self):
+    async def fetch_data(self):
         """Fetch the latest data from Phicomm server."""
+        session = self._hass.helpers.aiohttp_client.async_get_clientsession()
+
         if self._token is None:
             import hashlib
             md5 = hashlib.md5()
             md5.update(self._password.encode('utf8'))
-            data = {
-                'authorizationcode': AUTH_CODE,
-                'phonenumber': self._username,
-                'password': md5.hexdigest().upper()
-            }
+            data = {'authorizationcode': AUTH_CODE,
+                    'phonenumber': self._username,
+                    'password': md5.hexdigest().upper()}
             headers = {'User-Agent': USER_AGENT}
-            json = requests.post(TOKEN_URL, headers=headers, data=data).json()
+            async with session.post(TOKEN_URL, headers=headers, data=data) \
+                    as response:
+                json = await response.json(content_type=None)
+
             _LOGGER.debug("Get token: %s", json)
-            if 'access_token' in json:
-                self._token = json['access_token']
-                with open(self._token_path, 'w') as file:
-                    file.write(self._token)
-            else:
+            if 'access_token' not in json:
                 return None
+
+            self._token = json['access_token']
+            with open(self._token_path, 'w') as file:
+                file.write(self._token)
+
         headers = {'User-Agent': USER_AGENT, 'Authorization': self._token}
-        return requests.get(DATA_URL, headers=headers).json()
+        async with session.get(DATA_URL, headers=headers) as response:
```

Comment:
error handling

Reply:
I am not familar with Python. Just refer some other source code from ha release. They do the same code, so did I. Sorry for my non-professional code. If you can figure out the correct
 way, your help will be highly appreciated.

Reply:
You're already using try and exception handling in some parts, you just need it for the rest as well